cmd/soroban-rpc/internal/methods/simulate_transaction.go

```diff
@@ -199,9 +200,29 @@ func NewSimulateTransactionHandler(logger *log.Entry, networkPassphrase string,
 			}
 		}
 
+		// Get the auth data
+		var auth []string
+		if res.auth != nil {
+			// CGo doesn't have an easy way to do pointer arithmetic so,
+			// we are better off transforming the memory buffer into a large slice
+			// and finding the NULL termination after that
+			for _, a := range unsafe.Slice(res.auth, 1<<20) {
+				if a == nil {
```

Comment:
If auth is a null-terminated array of null-terminated strings, won't this hit the end of the first string then stop?

Reply:
(same question about the deallocator as well)

Reply:
This is Checking the top-level array, the null or each string is checked by `C.GoString()`

Reply:
i.e. doesn't the `unsafe.Slice(...)` return a slice like:
```
hi<null>bye<null><null>
```

Reply:
> i.e. I expect that the `unsafe.Slice(...)` of `["hi", "bye"]` would return a slice like:
> 
> ```
> hi<null>bye<null><null>
> ```

That’s not the format Rust (or rather, C) will return.

`["hi", "bye"]` will be returned from Rust as:

```
pointer_to("hi\0"), pointer_to("bye\0"),NULL
```

Reply:
Got it. Thanks.